### PR TITLE
Added the enpoint to create/update admin users.

### PR DESCRIPTION
**Pull request by seb-wolox**
## Summary

- I added the endpoint to create/update admin users.
- I implemented all the validations + tests + doc
- I made a few refactors (I tried to remove most `if ... else` statements in the new functions to make them more clear)

## Trello Card

https://trello.com/c/XDDnX5DG/17-usuario-administrador

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -2,22 +2,42 @@ const { users } = require('../models'),
...
   logger = require('../logger');
 
-const save = (request, response) =>
-  users
-    .create(request.body)
-    .then(newUser => {
-      const userData = newUser.dataValues;
-      delete userData.password;
+const save = async (request, response) => {
+  const user = Object.assign(request.body, { role: 'regular' });
```
Comment (gonziesc) [inline app/controllers/users.js]: use constants

File: app/controllers/users.js [hunk 2]
```diff
@@ -2,22 +2,42 @@ const { users } = require('../models'),
...
+
+  const userData = newUser.dataValues;
+  delete userData.password;
+
+  logger.info(`The user ${userData.firstName} has been created`);
+  response.status(201).send(userData);
+};
+
+const saveOrUpdateAdmin = async (request, response) => {
+  const user = Object.assign(request.body, { role: 'admin' });
```
Comment (gonziesc) [inline app/controllers/users.js]: same

File: app/middlewares/authorization.js [hunk 3]
```diff
@@ -36,3 +37,11 @@ module.exports = (request, response, next) => {
     response.status(401).json('Missing authorization token');
   }
 };
+
+const isAdmin = (request, response, next) => {
+  if (response.locals.loggedUser.role !== 'admin')
```
Comment (gonziesc) [inline app/middlewares/authorization.js]: same

File: test/users.js [hunk 4]
```diff
@@ -728,3 +730,576 @@ describe('/users GET', () => {
...
+
+  it('should update the user if it is already registered and update only the role', done => {
+    const validToken = jwt.encode(
+      { id: 3, email: 'noctis.lucis@wolox.com.ar' },
+      config.common.session.secret
+    );
+
+    const validUserId = 1;
+    const validUser = {
+      firstName: 'BLAHBLAHBLAH',
```
Comment (gonziesc) [inline test/users.js:771]: this is just for you to know that i read this :)

File: test/users.js [hunk 5]
```diff
@@ -728,3 +730,576 @@ describe('/users GET', () => {
...
+    };
+
+    chai
+      .request(server)
+      .post('/users/admin')
+      .set('x-access-token', validToken)
+      .send(validUser)
+      .catch(err => {
+        const res = err.response;
+        res.should.have.status(401);
```
Comment (gonziesc) [inline test/users.js]: big refactor: let's use just expect for everything. You can do it in another branch

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: test/users.js (hunk 4 above)
Comment (seb-wolox) [reply test/users.js:771]: Awesome! But, is there something wrong with that string?

File: test/users.js (hunk 5 above)
Comment (seb-wolox) [reply test/users.js]: No problem, it's just a RegExp.

File: test/users.js (hunk 4 above)
Comment (gonziesc) [reply test/users.js:771]: No!

File: app/helper/asyncHelper.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const to = promise => promise.then(data => [null, data]).catch(err => [err]);
```
Comment (gonziesc) [inline app/helper/asyncHelper.js:1]: I dont know if i like this but we can talk it later

**Review by gonziesc — APPROVED** [review]

**Pull request merged**